### PR TITLE
[4879] do not show empty space in sidebar explorer widgets when no children exist

### PR DESCRIPTION
**Pull request by jvega190**
- Allow user to go to an empty folder

https://github.com/craftercms/craftercms/issues/4879

### CODE REVIEW

File: ui/app/src/state/actions/pathNavigator.ts [hunk 1]
```diff
@@ -73,8 +73,11 @@ export const pathNavigatorFetchParentItems = /*#__PURE__*/ createAction<
   PayloadWithId<{ path: string; excludes?: string[]; limit: number; offset?: number; keyword?: string }>
 >('PATH_NAVIGATOR_FETCH_PARENT_ITEMS');
 
+export const pathNavigatorFetchPath =
+  /*#__PURE__*/ createAction<PayloadWithId<{ path: string }>>('PATH_NAVIGATOR_FETCH_PATH');
+
 export const pathNavigatorFetchPathComplete = /*#__PURE__*/ createAction<
-  PayloadWithId<{ parent?: DetailedItem; children: GetChildrenResponse }>
+  PayloadWithId<{ parent?: DetailedItem; children: GetChildrenResponse; path?: string }>
```
Comment (rart) [inline ui/app/src/state/actions/pathNavigator.ts]: What was path required for? Can't you get the path from `payload.parent`. It being conditional is not great so rather remove.

File: ui/app/src/state/reducers/pathNavigator.ts [hunk 2]
```diff
@@ -130,21 +131,26 @@ const reducer = createReducer<LookupTable<PathNavigatorStateProps>>(
       ...state,
       [payload.id]: { ...state[payload.id], isFetching: false, error: payload.error }
     }),
-    [pathNavigatorFetchPathComplete.type]: (state, { payload: { id, children } }) => {
-      const path = state[id].currentPath;
```
Comment (rart) [inline ui/app/src/state/reducers/pathNavigator.ts:134]: Revisit this (path should come from parent?)

**Review by rart — CHANGES_REQUESTED** [review]

File: ui/app/src/state/actions/pathNavigator.ts (hunk 1 above)
Comment (jvega190) [reply ui/app/src/state/actions/pathNavigator.ts]: I was able to get the path from `parent`. The only detail is that in `pathNavigatorChangePage` epic it wasn't sending the parent, but I was able to retrieve it from path navigator currentPath

File: ui/app/src/state/epics/pathNavigator.ts [hunk 3]
```diff
@@ -187,7 +211,13 @@ export default [
...
             offset
           }).pipe(
-            map((children) => pathNavigatorFetchPathComplete({ id, children })),
+            map((children) =>
+              pathNavigatorFetchPathComplete({
+                id,
+                children,
+                parent: state.content.itemsByPath[state.pathNavigator[id].currentPath]
+              })
+            ),
```
Comment (rart) [inline ui/app/src/state/epics/pathNavigator.ts]:
```suggestion
            map((children) => pathNavigatorFetchPathComplete({ id, children })),
```

File: ui/app/src/state/reducers/pathNavigator.ts [hunk 4]
```diff
@@ -130,12 +131,17 @@ const reducer = createReducer<LookupTable<PathNavigatorStateProps>>(
...
+    [pathNavigatorFetchPath.type]: (state, { payload }) => ({
+      ...state,
+      [payload.id]: { ...state[payload.id], isFetching: true, error: null }
+    }),
+    [pathNavigatorFetchPathComplete.type]: (state, { payload: { id, children, parent } }) => {
+      const path = parent?.path;
```
Comment (rart) [inline ui/app/src/state/reducers/pathNavigator.ts]:
```suggestion
    [pathNavigatorFetchPathComplete.type]: (state, { payload: { id, children, parent } }) => {
      const path = parent?.path ?? state[id].currentPath;
```

**Review by rart — CHANGES_REQUESTED** [review]

**Review by rart — APPROVED** [review]

**Pull request merged**